src/Firebase.cpp

```diff
@@ -41,45 +33,75 @@ String makeFirebaseURL(const String& path, const String& auth) {
 }  // namespace
 
 Firebase::Firebase(const String& host) : host_(host) {
-  http_.setReuse(true);
+  http_.reset(FirebaseHttpClient::create());
+  http_->setReuseConnection(true);
 }
 
 Firebase& Firebase::auth(const String& auth) {
   auth_ = auth;
   return *this;
 }
 
+const String& Firebase::auth() {
+  return auth_;
+}
+
 FirebaseGet Firebase::get(const String& path) {
-  return FirebaseGet(host_, auth_, path, &http_);
+  return FirebaseGet(host_, auth_, path, http_.get());
+}
+
+unique_ptr<FirebaseGet> Firebase::getPtr(const String& path) {
+  return unique_ptr<FirebaseGet>(new FirebaseGet(host_, auth_, path, http_.get()));
 }
 
 FirebaseSet Firebase::set(const String& path, const String& value) {
-  return FirebaseSet(host_, auth_, path, value, &http_);
+  return FirebaseSet(host_, auth_, path, value, http_.get());
+}
+
+unique_ptr<FirebaseSet> Firebase::setPtr(const String& path,
+                                         const String& value) {
+  return unique_ptr<FirebaseSet>(
+      new FirebaseSet(host_, auth_, path, value, http_.get()));
 }
 
 FirebasePush Firebase::push(const String& path, const String& value) {
-  return FirebasePush(host_, auth_, path, value, &http_);
+  return FirebasePush(host_, auth_, path, value, http_.get());
+}
+unique_ptr<FirebasePush> Firebase::pushPtr(const String& path, const String& value) {
+  return unique_ptr<FirebasePush>(
+      new FirebasePush(host_, auth_, path, value, http_.get()));
 }
 
 FirebaseRemove Firebase::remove(const String& path) {
-  return FirebaseRemove(host_, auth_, path, &http_);
+  return FirebaseRemove(host_, auth_, path, http_.get());
+}
+
+unique_ptr<FirebaseRemove> Firebase::removePtr(const String& path) {
+  return unique_ptr<FirebaseRemove>(
+      new FirebaseRemove(host_, auth_, path, http_.get()));
 }
 
 FirebaseStream Firebase::stream(const String& path) {
   // TODO: create new client dedicated to stream.
-  return FirebaseStream(host_, auth_, path, &http_);
+  return FirebaseStream(host_, auth_, path, http_.get());
+}
+
+unique_ptr<FirebaseStream> Firebase::streamPtr(const String& path) {
+  // TODO: create new client dedicated to stream.
+  return unique_ptr<FirebaseStream>(
+      new FirebaseStream(host_, auth_, path, http_.get()));
 }
 
 // FirebaseCall
 FirebaseCall::FirebaseCall(const String& host, const String& auth,
                            const char* method, const String& path,
-                           const String& data, HTTPClient* http) : http_(http) {
-  String url = makeFirebaseURL(path, auth);
-  http_->setReuse(true);
-  http_->begin(host, kFirebasePort, url, true, kFirebaseFingerprint);
+                           const String& data, FirebaseHttpClient* http) : http_(http) {
+  String path_with_auth = makeFirebaseURL(path, auth);
```

Comment:
in order part of the code with use camelCase, which style guide should we use? (ideally something that we can enforce with `clang-format`).